test/unit/test_chain_weight_clamp.cpp

```
@@ -0,0 +1,116 @@
+    SUBCASE("exactly the field maximum: still exact, not clamped off by one") {
+        CHECK(weight_of(seeds, 1, static_cast<int64_t>(MEM_CHAIN_W_MAX))
+              == static_cast<int>(MEM_CHAIN_W_MAX));
+    }
+
+    SUBCASE("over the maximum: saturates") {
+        // Two seeds of 2^26 each sum to 2^27, which is MEM_CHAIN_W_MAX + 1 --
+        // the smallest value the old (1<<30)-1 clamp let through to wrap, and it
+        // wrapped to 0.
+        const int over = weight_of(seeds, 2, static_cast<int64_t>(1) << 26);
+        CHECK(over == static_cast<int>(MEM_CHAIN_W_MAX));
+    }
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)